spec/section_spec.rb

```diff
@@ -41,6 +41,57 @@ class PageWithAnonymousSection < SitePrism::Page
           .with_message(error_message)
       end
     end
+
+    context 'default search arguments' do
+      class PageWithSectionWithDefaultSearchArguments < SitePrism::Page
+        class SectionWithDefaultArguments < SitePrism::Section
+          set_default_search_arguments :css, '.section'
+        end
+
+        section  :section_using_defaults, SectionWithDefaultArguments
+        section  :section_with_locator,   SectionWithDefaultArguments, '.other-section'
+        sections :sections,               SectionWithDefaultArguments
+      end
+      let(:page) { PageWithSectionWithDefaultSearchArguments.new }
+
+      context 'if default search arguments are not set' do
+        let(:search_arguments) { ['.other-section'] }
+
+        it 'should use arguments provided' do
+          expect(page).to receive(:find_first).with(*search_arguments).and_return(:element)
+          page.section_with_locator
+        end
+      end
+
+      context 'if default search arguments are set' do
+        let(:search_arguments) { [:css, '.section'] }
+
+        it 'should use arguments provided' do
+          expect(page).to receive(:find_first).with(*search_arguments).and_return(:element)
+          page.section_using_defaults
+        end
+      end
+
+      context 'if default search arguments are not set and no search arguments provided' do
+        it 'should raise ArgumentError' do
+          expect do
+            class ErroredPage < SitePrism::Page
+              section :section, Section
+            end
+          end.to raise_error(ArgumentError)
```

Comment:
can we pass a partial / full message into this.

Reply:
Will do